Libraries/LibWeb/HTML/SessionHistoryTraversalQueue.cpp

```diff
@@ -44,19 +50,28 @@ void SessionHistoryTraversalQueue::visit_edges(JS::Cell::Visitor& visitor)
 {
     Base::visit_edges(visitor);
     visitor.visit(m_queue);
+    visitor.visit(m_higher_priority_queue);
 }
 
-void SessionHistoryTraversalQueue::append(GC::Ref<GC::Function<void()>> steps)
+void SessionHistoryTraversalQueue::append(GC::Ref<GC::Function<void()>> steps, int priority)
 {
-    m_queue.append(SessionHistoryTraversalQueueEntry::create(vm(), steps, nullptr));
+    if (priority == 1) {
```

Comment:
You're not implementing actual _priority_ as in an int designating some relative weight to the individual tasks to execute, but rather a queue selection. I feel this would work better as an enum:

```
enum QueuePriority : u8 {
    High,
    Normal,
};
```